airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -533,24 +539,31 @@ def find_pod(self, namespace: str, context: Context, *, exclude_checked: bool =
 
         pod = None
         num_pods = len(pod_list)
-        if num_pods > 1:
-            raise AirflowException(f"More than one pod running with labels {label_selector}")
-        elif num_pods == 1:
+
+        if num_pods == 1:
             pod = pod_list[0]
-            self.log.info("Found matching pod %s with labels %s", pod.metadata.name, pod.metadata.labels)
-            self.log.info("`try_number` of task_instance: %s", context["ti"].try_number)
-            self.log.info("`try_number` of pod: %s", pod.metadata.labels["try_number"])
+            self.log_matching_pod(pod=pod, context=context)
+        elif num_pods > 1 and not self.reattach_on_restart:
+            self.log.warning("Found more than one pod running with labels %s, resolving ...", label_selector)
+            pod = self.process_duplicate_label_pods(pod_list)
+            self.log_matching_pod(pod=pod, context=context)
+        elif num_pods > 1:
+            raise IdenticalLabelPodError(f"More than one pod running with labels {label_selector}")
+
```

Comment:
For a slightly better readability, maybe you could rephrase the `elif` as:

```python

        elif num_pods > 1:
               if not self.reattach_on_restart:
                   self.log.warning("Found more than one pod running with labels %s, resolving ...", label_selector)
                   pod = self.process_duplicate_label_pods(pod_list)
                   self.log_matching_pod(pod=pod, context=context)
                else:
                   raise IdenticalLabelPodError(f"More than one pod running with labels {label_selector}")
```

Reply:
Rephrased with some improvements:)
Thank you for the suggestion @shahar1 !